sources/System/Timer/TimerSystem.cpp

```diff
@@ -4,22 +4,33 @@
 
 #include "TimerSystem.hpp"
 #include "Component/Timer/TimerComponent.hpp"
+#include "Drawables/2D/Text.hpp"
 
 using namespace std::chrono_literals;
+namespace RAY2D = RAY::Drawables::Drawables2D;
 
 namespace BBM
 {
 	TimerSystem::TimerSystem(WAL::Wal &wal)
 		: System(wal)
 	{}
 
-	void TimerSystem::onUpdate(WAL::ViewEntity<TimerComponent> &entity, std::chrono::nanoseconds dtime)
+	void TimerSystem::onUpdate(WAL::ViewEntity<TimerComponent, Drawable2DComponent> &entity, std::chrono::nanoseconds dtime)
 	{
 		auto &timer = entity.get<TimerComponent>();
 		timer.ringIn -= dtime;
 		if (timer.ringIn <= 0ns) {
 			timer.setDisable(true);
 			timer.callback(entity, this->_wal);
 		}
+
+		RAY2D::Text *text = dynamic_cast<RAY2D::Text *>(entity.get<Drawable2DComponent>().drawable.get());
```

Comment:
This is a huge no-no. You can't highjack a system to do something else. You should use the timer call to update this or create your own system to do that. With this modification, bombs won't explode.